src/datadog/remote_config/remote_config.cpp

```diff
@@ -30,24 +30,31 @@ constexpr std::array<uint8_t, sizeof(uint64_t)> capabilities_byte_array(
   return res;
 }
 
-Optional<product::Flag> parse_config_path(StringView config_path) {
+struct ConfigKeyData {
+  product::Flag product;
+  StringView config_id;
+};
+
+Optional<ConfigKeyData> parse_config_path(StringView config_path) {
   static const std::regex path_reg(
-      "^(datadog/\\d+|employee)/([^/]+)/[^/]+/[^/]+$");
+      "^(datadog/\\d+|employee)/([^/]+)/([^/]+)/[^/]+$");
 
   std::cmatch match;
   if (!std::regex_match(config_path.data(),
                         config_path.data() + config_path.size(), match,
                         path_reg)) {
-    return nullopt;
+    return {};
```

Comment:
I believe this change might not be necessary. Would it be possible to revert it?